### PR TITLE
Species types

### PR DESCRIPTION
**Pull request by rot4te**
Changed the types of Species fields to unitful numbers, eg charge::typeof(1u"q") and mass_in_eV::typeof(1.0u"eV/c^2")

### CODE REVIEW

File: src/species_initialize.jl [hunk 1]
```diff
@@ -173,9 +173,9 @@ function Species(name::String, charge::Int=0, iso::Int=-1)
                 spin = 0.5 * iso
             end
             if anti_atom == false
-                return Species(AS, charge, mass_in_eV, spin, 0, iso) # return the object to track
+                return Species(AS, charge*u"q", mass_in_eV*u"eV/c^2", spin*u"ħ", 0*u"eV/T", iso) # return the object to track
```
Comment (mattsignorelli) [inline src/species_initialize.jl:176]:
```suggestion
                return Species(AS, charge, mass_in_eV, spin, 0, iso) # return the object to track
```

File: src/species_initialize.jl [hunk 2]
```diff
@@ -173,9 +173,9 @@ function Species(name::String, charge::Int=0, iso::Int=-1)
...
             elseif anti_atom == true
-                return Species("anti-" * AS, charge, mass_in_eV, spin, 0, iso)
+                return Species("anti-" * AS, charge*u"q", mass_in_eV*u"eV/c^2", spin*u"ħ", 0u"eV/T", iso)
```
Comment (mattsignorelli) [inline src/species_initialize.jl:178]:
```suggestion
                return Species("anti-" * AS, charge, mass_in_eV, spin, 0, iso)
```

Comment (rot4te) [reply src/species_initialize.jl:178]: Unfortunately that doesn't work in testing. Once I changed the types of the struct, I need to give them units.

File: src/set_units.jl [hunk 3]
```diff
@@ -309,7 +309,7 @@ function mass(species::Species, unit::Union{Unitful.FreeUnits,AbstractString}=ge
   if dimension(unit) != dimension(u"kg")
     throw(ErrorException("unit have proper dimension"))
   end
-  return (species.mass_in_eV * u"eV/c^2" |> unit).val
+  return (species.mass_in_eV |> unit).val
```
Comment (mattsignorelli) [inline src/set_units.jl:312]: Both `mass` and `charge` should be switched to `massof` and `chargeof`. And if `nothing` is passed, then use `upreferred`. E.g.

```julia
function massof(species::Species, unit::Union{Nothing, Unitful.FreeUnits}=nothing)
  if isnothing(unit)
    return upreferred(species.mass)
  else
    return uconvert(species.mass, unit)
  end
end
```
Likewise for `chargeof`

File: src/set_units.jl [hunk 4]
```diff
@@ -335,6 +335,6 @@ function charge(species::Species, unit::Union{Unitful.FreeUnits,AbstractString}=
   if dimension(unit) != dimension(u"C")
     throw(ErrorException("unit have proper dimension"))
   end
-  return (species.charge * u"e" |> unit).val
+  return (species.charge |> unit).val
```
Comment (mattsignorelli) [inline src/set_units.jl:338]: See comments for `mass`

**Review by mattsignorelli — CHANGES_REQUESTED** [review]
See the comments near `mass` and `charge`, those functions need minor changes and removal of the unnecessary unit conversion of `Species` for default constructor

File: src/species_initialize.jl (hunk 2 above)
Comment (mattsignorelli) [reply src/species_initialize.jl:178]: interesting, i'll take a look later

Comment (mattsignorelli) [reply src/species_initialize.jl:178]: ok yes i get error about dimensional compatbility

**Comment by mattsignorelli** [comment]
@lllx125 implemented `massof`, `chargeof`, and `setunits` all correct, and is merging into main momentarily. However his `Species` struct still has non-unitful quantities. Could you pull the latest main and resolve the merge conflicts?

**Pull request closed without merging**